server/lib/chat/chat-api.ts

```diff
@@ -184,6 +199,28 @@ export class ChatApi {
     return await this.chatService.joinChannel(channelName, ctx.session!.userId)
   }
 
+  @httpPatch('/:channelId')
+  @httpBefore(throttleMiddleware(editThrottle, ctx => String(ctx.session!.userId)))
+  async editChannel(ctx: RouterContext): Promise<EditChannelResponse> {
+    const channelId = getValidatedChannelId(ctx)
+    const { body } = validateRequest(ctx, {
+      body: Joi.object<EditChannelRequest>({
+        description: Joi.string().allow(null),
+        topic: Joi.string().allow(null),
+        bannerId: Joi.string().uuid(),
+      }),
+    })
+
+    if (body.description === 'null') {
```

Comment:
We probably need to do something better about this. I was fine with this approach with leagues because it was admin-only but this is not, and thus we're blocking people from setting the string 'null' as a topic or description, which seems pretty dumb